### PR TITLE
[RFC][AMDGPU] Remove old llvm.amdgcn.buffer.* and tbuffer intrinsics

### PR DESCRIPTION
**Pull request by jayfoad**
They have been superseded by llvm.amdgcn.raw.buffer.* and
llvm.amdgcn.struct.buffer.*.

### CODE REVIEW

**Comment by jayfoad** [comment]
@llvm/pr-subscribers-mlir-amdgpu I see the MLIR build failed with:
```
mlir/include/mlir/Dialect/LLVMIR/ROCDLConversions.inc:142:28: error: no member named 'amdgcn_buffer_load' in namespace 'llvm::Intrinsic'; did you mean 'amdgcn_s_buffer_load'?
mlir/include/mlir/Dialect/LLVMIR/ROCDLConversions.inc:151:28: error: no member named 'amdgcn_buffer_store' in namespace 'llvm::Intrinsic'; did you mean 'amdgcn_raw_buffer_store'?
```
Does MLIR really depend on these old intrinsics? Can it be fixed?

**Comment by jayfoad** [comment]
@marekolsak does Mesa use any of the old buffer intrinsics?

**Comment by krzysz00** [comment]
@jayfoad MLIR does not meaningfully depend on those intrinsics.

No current lowering uses the wrappers we have around those intrinsics.

Those wrappers are defined at https://github.com/llvm/llvm-project/blob/main/mlir/include/mlir/Dialect/LLVMIR/ROCDLOps.td#L428-L464 and are the same sort of "legacy infrastructure, do not use" as the intrinsics you're planning to remove in this PR.

You should be able to just delete the highlighted bit and any code that tries to refer to those (mainly their parsers and printers over in `mlir/lib/Dialect/LLVMIR/IR/ROCDLDialect.cpp` and the tests for the translation of those operations.

Also, re the migration, I'd suggest people might wart to move to the `raw.ptr.*` versions, since they're the newest and best-behaved ones.

**Comment by marekolsak** [comment]
> @marekolsak does Mesa use any of the old buffer intrinsics?

No, Mesa doesn't use them.

**Comment by jayfoad** [comment]
Ping.

**Review by arsenm — APPROVED** [review]

File: llvm/test/CodeGen/AMDGPU/llvm.amdgcn.atomic.fadd.gfx90a.ll [hunk 1]
```diff
@@ -1,93 +0,0 @@
-; RUN: llc < %s -mtriple=amdgcn -mcpu=gfx90a -verify-machineinstrs | FileCheck %s -check-prefix=GFX90A
-; RUN: not --crash llc < %s -mtriple=amdgcn -mcpu=gfx908 -verify-machineinstrs 2>&1 | FileCheck %s -check-prefix=GFX908
-
-declare float @llvm.amdgcn.buffer.atomic.fadd.f32(float, <4 x i32>, i32, i32, i1)
-declare <2 x half> @llvm.amdgcn.buffer.atomic.fadd.v2f16(<2 x half>, <4 x i32>, i32, i32, i1)
-declare float @llvm.amdgcn.global.atomic.fadd.f32.p1.f32(ptr addrspace(1), float)
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/llvm.amdgcn.atomic.fadd.gfx90a.ll:6]: This deleted a bunch of not-buffer atomic tests

Comment (arsenm) [reply llvm/test/CodeGen/AMDGPU/llvm.amdgcn.atomic.fadd.gfx90a.ll:6]: We're missing run lines in other tests, I'll deal with this

**Pull request merged**